internal/prober/http/http.go

```diff
@@ -21,18 +21,26 @@ import (
 	"github.com/rs/zerolog"
 )
 
+const checkProbeIdHeader = "x-sm-id"
+
 var errUnsupportedCheck = errors.New("unsupported check")
 
 type Prober struct {
 	config                     config.Module
 	cacheBustingQueryParamName string
 }
 
-func NewProber(ctx context.Context, check sm.Check, logger zerolog.Logger) (Prober, error) {
+func NewProber(ctx context.Context, check sm.Check, logger zerolog.Logger, checkProbeIdentifier string) (Prober, error) {
```

Comment:
Because we might want to add the `x-sm-id` header to other check types, for example `multihttp`, I wonder if it would make sense to move the constant and header definition to `internal/prober/prober.go` instead (or even in a separate `header.go` file)? In that case we could modify the `NewProber` constructor here to accept something more generic such as `reservedHeaders/systemHeaders http.Header`.

Reply:
Liked this a lot -- was initially looking for a way to make the input more generic. Thanks for the reference to `net.http.Header`!